Doc/tools/extensions/pyspecific.py

```diff
@@ -495,13 +500,13 @@ def write(self, *ignored):
             self.topics[label] = writer.output
 
     def finish(self):
-        f = open(path.join(self.outdir, 'topics.py'), 'wb')
-        try:
-            f.write('# -*- coding: utf-8 -*-\n'.encode('utf-8'))
-            f.write(('# Autogenerated by Sphinx on %s\n' % asctime()).encode('utf-8'))
-            f.write(('topics = ' + pformat(self.topics) + '\n').encode('utf-8'))
-        finally:
-            f.close()
+        fn = path.join(self.outdir, 'topics.py')
+        with open(fn, 'w', encoding='utf-8') as f:
+            write = partial(print, file=f)
```

Comment:
Is using a partial function here worth it?  I find `print` and `pprint` convenient with their `file`/`stream` parameter 🙂